plugins/tech-insights-backend/migrations/2022060100821_facts_timestamp_precision.js

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-check
+
+/**
+ * @param {import('knex').Knex} knex
+ */
+exports.up = async function up(knex) {
+  await knex.schema.alterTable('facts', table => {
+    table
+      .dateTime('timestamp', { precision: 0 })
+      .defaultTo(knex.fn.now())
+      .notNullable()
+      .comment('The timestamp when this entry was created')
+      .alter();
```

Comment:
Hmm. I think this is one of those times when knex wants a `.alter({ alterType: true })` here 🤔

Reply:
AFAIK that should be the default value for this method call: https://github.com/knex/knex/blob/a4df3e35403756615f297ef75ac1d998035d65e6/lib/schema/columnbuilder.js#L127

Confirmed that the modification gets applied correcly on my pg 12.5